### PR TITLE
Public Dashboards: Disable email-sharing when there is no license

### PR DESCRIPTION
**Pull request by AgnesToulet**
**What is this feature?**
Disable email-shared dashboards when the feature is disabled. 

**Why do we need this feature?**
Currently, these dashboards are displayed as "public" but they are only email-shared and the whole feature should be disabled when there is no license. These dashboards should be not displayed at all.

**Who is this feature for?**
Enterprise users who removed their licenses. 

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana-enterprise/issues/6026

**Special notes for your reviewer:**
Enterprise PR: https://github.com/grafana/grafana-enterprise/pull/6184

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: docs/sources/administration/enterprise-licensing/_index.md [hunk 1]
```diff
@@ -196,6 +196,12 @@ The active users limit is turned off immediately.
 
 Settings updates at runtime are not affected by an expired license.
 
+#### Email sharing
+
+Email-shared dashboards won't be available for external users anymore. These dashboards will become private again but you will be able to make them public to everybody if you want to.
+
+Configurations will be kept and restored if you update your license.
+
```
Comment (jdbaldry) [inline docs/sources/administration/enterprise-licensing/_index.md:204]: I've tried to rewrite the documentation to use present tense and in active voice to be more consistent with the Grafana Labs style guide in https://grafana.com/docs/writers-toolkit/.

Let me know what you think, or use feel free to apply the suggestion if it is technically accurate and still makes sense.

```suggestion
#### Email sharing

External users can't access dashboards shared via email.
These dashboards are now private but you can make them public and accessible to everyone if you want to.

Grafana keeps your email configurations and restores them after you update your license.

```

**Review by evictorero — APPROVED** [review]
Tested several cases and works as expected!

File: docs/sources/administration/enterprise-licensing/_index.md (hunk 1 above)
Comment (AgnesToulet) [reply docs/sources/administration/enterprise-licensing/_index.md:204]:
```suggestion
#### Email sharing

External users can't access dashboards shared via email anymore.
These dashboards are now private but you can make them public and accessible to everyone if you want to.

Grafana keeps your sharing configurations and restores them after you update your license.

```

@jdbaldry What about this? I'm afraid "email configurations" can be confusing.

Comment (jdbaldry) [reply docs/sources/administration/enterprise-licensing/_index.md:204]: Yeah great proposal! I think you are right that "email configurations" is a bit confusing, "sharing configurations" is much better :)

**Review by jdbaldry — APPROVED** [review]
Docs look good

**Review by juanicabanas — APPROVED** [review]
LGTM!

**Pull request merged**